pkg/pod/container.go

```diff
@@ -113,6 +113,91 @@ func (builder *ContainerBuilder) WithSecurityCapabilities(sCapabilities []string
+//
+//nolint:gocognit
+func (builder *ContainerBuilder) WithDropSecurityCapabilities(sCapabilities []string, redefine bool) *ContainerBuilder {
+	glog.V(100).Infof("Dropping a list of SecurityCapabilities %v from container %s",
+		sCapabilities, builder.definition.Name)
+
+	if !areCapabilitiesValid(sCapabilities) {
+		glog.V(100).Infof("Given SecurityCapabilities %v are not valid. Valid list %s",
+			sCapabilities, AllowedSCList)
+
+		builder.errorMsg = "one of the give securityCapabilities is invalid. Please extend allowed list or fix parameter"
+	}
+
+	var sCapabilitiesList []v1.Capability
+	for _, capability := range sCapabilities {
+		sCapabilitiesList = append(sCapabilitiesList, v1.Capability(capability))
+	}
+
+	// filter possible duplicated capabilities from user's input
+	sCapabilitiesList = uniqueCaps(sCapabilitiesList)
+
+	// filter conflicting caps between ADD and DROP
+	if builder.definition.SecurityContext != nil {
+		if builder.definition.SecurityContext.Capabilities != nil {
+			if builder.definition.SecurityContext.Capabilities.Add != nil {
+				sCapabilitiesList = capsDifference(
+					builder.definition.SecurityContext.Capabilities.Add,
+					sCapabilitiesList, false)
+
+				if len(sCapabilitiesList) > 0 {
+					glog.V(100).Infof("Conflicting ADD and DROP capabilities")
+
+					for _, mcap := range sCapabilitiesList {
+						glog.V(100).Infof("SecurityCapability %q already present in the Capabilities.Add list", mcap)
+					}
+
+					builder.errorMsg = "Conflicting ADD and DROP SecurityCapabilities"
+				}
+			}
+		}
+	}
+
+	if builder.errorMsg != "" {
+		return builder
+	}
+
+	if builder.definition.SecurityContext != nil {
+		if builder.definition.SecurityContext.Capabilities != nil {
+			if builder.definition.SecurityContext.Capabilities.Drop != nil {
+				if !redefine {
+					sCapabilitiesList = capsDifference(
+						builder.definition.SecurityContext.Capabilities.Drop,
+						sCapabilitiesList, true)
+
+					glog.V(100).Infof("Updating existing SecurityContext.Capabilities.Drop list")
+					builder.definition.SecurityContext.Capabilities.Drop = append(
+						builder.definition.SecurityContext.Capabilities.Drop,
+						sCapabilitiesList...)
+				} else {
+					glog.V(100).Infof("Resetting existing SecurityContext.Capabilities.Drop list")
+					builder.definition.SecurityContext.Capabilities.Drop = sCapabilitiesList
+				}
+			} else {
+				glog.V(100).Infof("Setting SecurityContext.Capabilities.Drop list")
+				builder.definition.SecurityContext.Capabilities.Drop = sCapabilitiesList
+			}
+		} else {
+			glog.V(100).Infof("Defining new SecurityContext.Capabilities")
+			builder.definition.SecurityContext.Capabilities = &v1.Capabilities{
+				Drop: sCapabilitiesList,
+			}
+		}
+	} else {
+		glog.V(100).Infof("Defining new SecurityContext with Capabilities.Drop")
+		builder.definition.SecurityContext = &v1.SecurityContext{
+			Capabilities: &v1.Capabilities{
+				Drop: sCapabilitiesList,
+			},
+		}
+	}
```

Comment:
Nesting conditional this far is a bit hard to follow. Would something like this work for you?
```
	if builder.definition.SecurityContext == nil {
		builder.definition.SecurityContext = new(v1.SecurityContext)
	}

	if builder.definition.SecurityContext.Capabilities == nil {
		builder.definition.SecurityContext.Capabilities = new(v1.Capabilities)
	}

	if !redefine {
		sCapabilitiesList = capsDifference(
			builder.definition.SecurityContext.Capabilities.Drop,
			sCapabilitiesList, true)

		glog.V(100).Infof("Updating existing SecurityContext.Capabilities.Drop list")
		builder.definition.SecurityContext.Capabilities.Drop = append(
			builder.definition.SecurityContext.Capabilities.Drop,
			sCapabilitiesList...)
	} else {
		glog.V(100).Infof("Resetting existing SecurityContext.Capabilities.Drop list")
		builder.definition.SecurityContext.Capabilities.Drop = sCapabilitiesList
	}
```

Reply:
thanks, updated